### PR TITLE
Archive Carousel for desktop and mobile

### LINKED ISSUE #73: Create archive section and carousel
Create the archive section with a carousel that can be clicked through

For right now, we are just looking for the functionality of this, not necessarily the information. Some examples will be listed below.

DoD:
- There is a carousel that rotates through images, titles, descriptions, and links every 5 seconds
- The following pictures/description/titles/link are available
- It should span the width of the page
- The user should be able to click through them
- It should use proper font sizing, coloring
- The background image should be darkened
- The images should be locally available

Example:
Title: Flourish! 2019
Description: Flourish 2019 was cancelled due to low interest and insignificant marketing. 
Link: flourishconf.com
Picture:
image

Title: Flourish! 2018
Description: Flourish 2018
Link: flourishconf.com
Picture:
image

### LINKED ISSUE #74: Create archive section and carousel mobile
Create the archive section with a carousel automatically rotates through

For right now, we are just looking for the functionality of this, not necessarily the information. Some examples will be listed below.

DoD:
- There is a carousel that rotates through images, titles, descriptions, and links every 5 seconds
- The following pictures/description/titles/link are available
- It should span the width of the page
- The user should be able to click through them
- It should use proper font sizing, coloring
- The background image should be darkened
- The images should be locally available

Example:
Title: Flourish! 2019
Description: Flourish 2019 was cancelled due to low interest and insignificant marketing. 
Link: flourishconf.com
Picture:
image

Title: Flourish! 2018
Description: Flourish 2018
Link: flourishconf.com
Picture:
image

### PR DESCRIPTION
**Pull request by rapterjet2004**
Fixes #73 
Fixes #74 

Sorry for the late PR, had midterms all of last week. Decided to knock out two birds with one stone on this one. 

- There is a carousel that rotates through images, titles, descriptions, and links every 5 seconds
- The following pictures/description/titles/link are available
- It should span the width of the page
- The user should be able to click through them
- It should use proper font sizing, coloring
- The background image should be darkened
- The images should be locally available
- Responsive, carousel css adjusts for mobile devices (I made sure to use the same bounds as tailwindcss, which is 640px width to indicate small devices. I didn't include anything for tablet screens b/c the format looked alright as is)
- Scalable, to add a new slide, simply add a new image to /assets/images/archiveImages, and then modify the list in ArchiveCarousel/allarchive.jsx to match and the code does the rest
- I also downloaded two (free and transparent) chevron arrow icons for the carousel and included them under assets/icons/
- Added the Carousel to the about and social pages
- Also went ahead and made the WhoWeAreSection a component. 

**Note**: 
Also, I know that the carousel's formatting isn't correct on mobile, but I don't believe that's an error with my code as it effects all components in the about page. I did a quick test of #78 and I believe it fixes the error.

### CODE REVIEW

File: src/components/Description/index.jsx [hunk 1]
```diff
@@ -1,7 +1,7 @@
 
 export default function Description(){
     return (<div className={"container m-auto px-[5rem]"}>
-                    <div className={"mx-[5rem] min-w-full"}>
+                    <div className={"text-center sm:mx-[5rem] sm:min-w-full sm:text-left"}>
```
Comment (rapterjet2004) [inline src/components/Description/index.jsx:4]: Don't know why this is showing up as changed, I reverted my prior commit, this should be unchanged file

File: src/pages/about.js [hunk 2]
```diff
@@ -2,20 +2,17 @@ import './about.css'
 import SpeakersSection from '../components/SpeakersSection';
 import ExecutiveLeaders from '../components/ExecutiveLeaders';
 import Description from '../components/Description';
+import ArchiveCarousel from '../components/ArchiveCarousel';
+import WhoWeAreSection from '../components/WhoAreWeSection';
```
Comment (ChristianBingman) [inline src/pages/about.js:6]: Could you call it WhoAreWe instead of WhoAreWeSection

File: src/components/ArchiveCarousel/index.jsx [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    return () => {
+      if (interval) {
+        clearInterval(interval);
+      }
+    }
+  })
+
+  return (
+    <div className="carousel" >
+      <h1 class="grid place-items-center">Archive</h1>
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/index.jsx:35]: Class is invalid JSX, use classname

File: src/components/ArchiveCarousel/index.jsx [hunk 4]
```diff
@@ -0,0 +1,61 @@
...
+      <button className="indicatorLeft" onClick={() => {
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/index.jsx:36]: Could you remove the indicators on mobile?

File: src/components/ArchiveCarousel/archivecarousel.css [hunk 5]
```diff
@@ -0,0 +1,29 @@
+.carousel{
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/archivecarousel.css:1]: Most of these can be implemented in tailwind

File: src/components/ArchiveCarousel/Item/index.jsx [hunk 6]
```diff
@@ -0,0 +1,26 @@
...
+    window.matchMedia("(min-width: 640px)").matches
+  )
+
+  useEffect( ()=> {
+    window.matchMedia("(min-width: 640px)").addEventListener('change', e=> setMatches(e.matches));
+  })
+
+    return (
+      <div  class="carousel-item" style={{width: "100%"}}>
+        {matches && (<h1 class="title" >{props.title}</h1>)}
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/Item/index.jsx:16]: class is invalid jsx

File: src/components/ArchiveCarousel/Item/item.css [hunk 7]
```diff
@@ -0,0 +1,62 @@
...
+    word-wrap: break-word;
+    color: white;
+    z-index: 1;
+}
+
+.carousel-item .link-mobile {
+    position: absolute;
+    font-size: .6rem;
+    top: 75%;
+    color: yellow;
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/Item/item.css:60]: The link color is secondary dark iirc

**Review by ChristianBingman — CHANGES_REQUESTED** [review]
Needs a little more changes, few major notes, class is invalid in react since class is reserved in JS, but you are writing inline HTML essentially. I will be upping the estimate on this, I really appreciate all the work on this!

**Comment by rapterjet2004** [comment]
Got it, I
- deleted the css files and used tailwind instead
- changed the website link color to secondary dark
- changed all class keywords to className
- renamed WhoAreWeSection to just WhoAreWe

File: src/pages/about.js (hunk 2 above)
Comment (ChristianBingman) [reply src/pages/about.js]: It should be WhoAreWe, much better for consistency.

**Review by ChristianBingman — CHANGES_REQUESTED** [review]
Very very close, could you fix the naming of WhoAreWe for consistency, and there was a new error in the [build](https://github.com/acm-uic/flourish-2023/actions/runs/3401983003/jobs/5657440283#step:3:43) (You can ignore the prop-type errors for now.

File: src/components/ArchiveCarousel/index.jsx [hunk 8]
```diff
@@ -46,7 +46,7 @@ export default function ArchiveCarousel() {
       <div className="whitespace-nowrap	transition duration-500" style={{transform: `translateX(-${index*100}%)`}}>
         {
           ARCHIVE.map((c) => {
-            return <Item name={c[0]} title={c[1]} description={c[2]} link={c[3]}/>
+            return <Item key={c} name={c[0]} title={c[1]} description={c[2]} link={c[3]}/>
```
Comment (ChristianBingman) [inline src/components/ArchiveCarousel/index.jsx:49]: Not a great key, but that'll work

**Review by ChristianBingman — APPROVED** [review]
Great thanks!

**Pull request merged**